docs/specs/editorconfig-in-the-compiler.md

```diff
@@ -0,0 +1,111 @@
+All-capital words are interpreted per [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+# Compiler Command Arguments
+
+The compiler *MUST* receive a new switch, `/editorconfig`, which accepts a single
+file which is an .editorconfig file that may apply to one or more files in the
+compilation. Multiple files are passed with multiple `/editorconfig` switches,
+the same way you can pass multiple references with /reference or multiple analyzers
+with /analyzer. There is no short form of the argument. Files passed with relative
```

Comment:
> I do not believe we need to prevent the search of parent directories for .editorconfig files by default. 

I **strongly** feel this important.  To the point it's simply a non-starter for the compiler.  Ambient authorities on the file system affecting the success / failure of a compilation is not a feature I want to sell to management.  Nor is it the type of problem I want to get endless "compiler doesn't work" bug reports about. 

> Specifically, no properties previously defined on editorconfig.org affect which warnings are reported or the severity with which they are reported during a build

This is something I've specifically brought up with @jasonmalinowski.  If the eventual goal is to enable this feature then it must be dealt with in V1.  Even if the feature is to recognize the syntax and fail compilation overall due to "this is not yet implemented" is acceptable.  

In V2 giving new meaning to existing entries in editorconfig is a breaking change.  It's tantamount to introducing new warnings for existing code.  

The idea that the syntax isn't valid today is a mitigation, but not a strong one.  True it's a strong one if you consider a single repo and it's evolution of editorconfig and compilers independently from everything else.  In that case sure you're very rarely going to get tripped up by adding a magic warning syntax before using the compiler that leverages it.  

But code rarely evolves in isolation.  More commonly it evolves by taking bits and pieces from other projects.  That means you can very easily end up with the following:

1. Repo1: using editorconfig and C# 8.0 where pragma disables are supported by the compiler. 
1. Repo2: using C# 7.1 

Repo 2 decides to folllow the style of Repo 1 and snags the editorconfig.  A year later it updates to C# 8.0 and I get a bug report saying "upgrading to C# 8.0 broke my build".  

> When .editorconfig is added to a specific project with root=true, searching to the drive root by default is not a problem.

I have 0 problem with considering existing .editorconfig in the repo.  It's very sensible.  The compiler by default searching the entire machine for .editorconfig entries when the developer never opted into the feature though is not something I'm in favor of.